doc/tutorials/content/sources/custom_executor/custom_executor.cpp

```diff
@@ -0,0 +1,81 @@
+#include <pcl/common/generate.h>
+#include <pcl/filters/functor_filter.h>
+#include <pcl/visualization/cloud_viewer.h>
+#include <chrono>
+
+namespace executor
+{
+// Forward declaration for custom executor
+template <typename Blocking, typename ProtoAllocator>
+struct omp_benchmark_executor;
+
+// Mark the executor as available
+template <>
+struct is_executor_available<omp_benchmark_executor> : std::true_type
+{
+};
+
+// Custom executor derived from OMP executor
+template <typename Blocking = blocking_t::always_t, typename ProtoAllocator = allocator_t<void>>
+struct omp_benchmark_executor : omp_executor<Blocking, ProtoAllocator>
+{
+  // Introduce base struct members
+  using Base = omp_executor<Blocking, ProtoAllocator>;
+  using Base::max_threads;
+  using typename Base::index_type;
+  using typename Base::shape_type;
+
+  template <typename F>
+  void bulk_execute(F &&f, const shape_type &n) const
+  {
+    // Throw static assert failure if executor is not available
+    static_assert(is_executor_available_v<omp_executor>, "OpenMP benchmark executor unavailable");
+
+    // Limit the the number of threads, to the maximum set in the executor
+    const auto num_threads = n ? std::min(max_threads, n) : max_threads;
+
+    // Measure total time taken by all threads
+    auto total_t1 = std::chrono::steady_clock::now();
```

Comment:
`const` as much as possible (lots of places)

Reply:
Done.